app/coffeescripts/views/conferences/ConferenceView.coffee

```diff
@@ -126,3 +128,83 @@ define [
         else
           window.open(data[0].url)
       )
+
+    deleteRecording: (e) ->
+      e.preventDefault()
+      if !confirm I18n.t("Are you sure you want to delete this recording?")
+        $(e.currentTarget).parents('.inline-block').find('.al-trigger').focus()
+        return
+      $deleteButton = $(e.currentTarget).parent()
+      @toggleActionButton($deleteButton, {state: "processing", action: "delete"})
+      @toggleRecordingLink($deleteButton, {state: "processing"})
+      $.ajaxJSON($deleteButton.data('url') + "/recording", "DELETE", {
+          recording_id: $deleteButton.data("id"),
+        }, (data) =>
+          if data.deleted == "true"
+            @removeRecordingRow($deleteButton)
+            return
+          @ensureActionPerformed($deleteButton, 1, @removeRecordingRow)
+      )
+
+    ensureActionPerformed: ($button, attempt, callback) =>
+      $.ajaxJSON($button.data('url') + "/recording", "GET", {
+          recording_id: $button.data("id"),
+        }, (data) =>
+          if $.isEmptyObject(data)
+            callback($button)
+            return
+          if attempt <= 5
+            attempt++
+            setTimeout((=> @ensureActionPerformed($button, attempt, callback); return;), attempt * 1000)
+            return
```

Comment:
So if I understand this you are going to loop 5 times ensuring that we actually deleted until its actually deleted? Why is this necessary? it should just be one you are going to kill the server if we hit this 5 times if something fails.

Reply:
First lets go to the reasons behind. Why ajax and no a direct request to the controller as the delete conference action?

As the recordings live in an external server (lets say BBB), the action deleteRecordings can fail. 

But in fact, the real issue is that this was required to workaround a delay in BBB for executing some of the actions. 

Until 4 months ago (the original version of this code was written more than 1 year ago), there was a delay when performing some actions on recordings. Which means that when the action was sent to BBB it was put in a queue and the BBB API doesn't support callbacks. As the action was not made effective immediately, performing a direct request to the controller would have lead to a new query for fetching the recordings from BBB resulting in a race condition that may have as a response a recording in the state it was before the user clicked the button. From the user perspective, the application was failing.

Doing it with ajax and performing an extra (or a few) requests to confirm that the expected state was actually reached is highly recommended (from us) to anyone doing integrations with BBB.

Now, why 5 times. It is really an arbitrary number. It could be 1 it could be 3, it didn't really matter as long as there was a mechanism for confirmation in place. This implementation performs the first verification after 1 sec, the second after two and so on. Usually the first one would give the correct state, but sometimes 3. 5 was more like to play 'safe'.

Now, putting this in context. Management action on recordings can be done only by the conference owners and administrators. So the possibility of over killing a server with multiple requests is really low, as not even them are performing deleteRecording all the time. 

To be honest, we may remove the verification now as the action is expected to be executed immediately. But I kept it because the recording still lives in an external server and there is always a chance of the action to fail. And as BBB doesn't have callbacks, well the only way to verify if the action was performed is to perform the second request.

So, what should we do?

Reply:
Alright, yes and no.
All the request are going to BBB through Canvas
Client->Canvas->BBB
The delete is requested only once.
deleteRecording (Client)->recording DELETE (Canvas controller)->deleteRecordings (BBB API)

The second request (therefore the one that is repeated) pull the current status.
ensureActionPerformed (Client)->recording GET (Canvas controller)->getRecordings (BBB API)

But is cool, lets change the approach. As the delete (and all the other action on recordings) are now executed immediately in BBB (no more queue for processing) I am removing the code that does the verification.